source/DSCResources/MSFT_ADDomain/MSFT_ADDomain.psm1

```diff
@@ -541,6 +565,16 @@ function Set-TargetResource
             Write-Verbose -Message ($script:localizedData.CreatedForest -f $DomainName)
         }
 
+        $needsReboot = $true
+    }
+    elseif(
+        [string]::IsNullOrEmpty($targetResource.SysvolPath) -and
+        (Test-Path -Path 'HKLM:\SYSTEM\CurrentControlSet\Services\Netlogon\Parameters\LocatorDCPromoPreRebootHint')
+    ) {
+        $needsReboot = $true
+    }
+
+    if ($needsReboot -and -not (Get-Process -Name dsc -ErrorAction SilentlyContinue | Where-Object { $_.Path -match 'dsc\.exe$' })) {
```

Comment:
I'm not liking this addition. A dsc process can be running for different reasons, it could be another process that is called dsc.exe that has nothing to do with DSCv3. We need to have better support in DSCv3 for rebooting.

Reply:
Suggest adding a new optional parameter `SuppressRestart` so a user can control if the resource should reboot or not.